### PR TITLE
srml-assets: AssetID should be generic parameter

### PR DESCRIPTION
**Pull request by fzzr-**
Improved srml-assets: AssetId is generic type-parameter now.

Needed because we want use `u8`/`i128`/Hash/_custom wrapper-struct_ as `AssetID`.

### CODE REVIEW

**Comment by fzzr-** [comment]
[clabot:check]

File: srml/assets/src/lib.rs [hunk 1]
```diff
@@ -145,7 +147,7 @@ decl_module! {
 			let origin = ensure_signed(origin)?;
 
 			let id = Self::next_asset_id();
-			<NextAssetId<T>>::mutate(|id| *id += 1);
+			<NextAssetId<T>>::mutate(|id| *id += 1.into());
```
Comment (bkchr) [inline srml/assets/src/lib.rs]:
```suggestion
			<NextAssetId<T>>::mutate(|id| *id += One::one());
```

You will need to import `One`.

**Comment by bkchr** [comment]
Overall looking okay, just some nitpicks.

File: srml/assets/src/lib.rs [hunk 2]
```diff
@@ -131,9 +132,10 @@ pub trait Trait: system::Trait {
 
 	/// The units in which we record balances.
 	type Balance: Member + Parameter + SimpleArithmetic + Default + Copy;
-}
 
-type AssetId = u32;
+	/// The arithmetic type of asset identifier.
+	type AssetId: Parameter + SimpleArithmetic + Default + Copy + Encode + Decode;
```
[the remark this answers is not in the document]
Comment (fzzr-) [reply srml/assets/src/lib.rs]: You are right, thanks.

File: srml/assets/src/lib.rs (hunk 1 above)
Comment (fzzr-) [reply srml/assets/src/lib.rs]: it's more explicit way. Thanks.

**Comment by fzzr-** [comment]
Probably something missed? I don't understanding why [CI task failing](https://gitlab.parity.io/parity/substrate/-/jobs/160333). Locally all test passed.

**Comment by bkchr** [comment]
You can ignore `check-runtime`. `srml-assets` is not included in the runtime and thus we don't need to increase the spec or impl version.

**Comment by fzzr-** [comment]
@bkchr, I fix all your review suggestions and so it can be merged.

**Comment by bkchr** [comment]
@fzzr- Could you please merge master that we get the CI happy? :) There was some issue, but nothing related to your pr :)

**Pull request merged**